src/H5B2int.c

```diff
@@ -139,7 +139,7 @@ H5B2__split1(H5B2_hdr_t *hdr, uint16_t depth, H5B2_node_ptr_t *curr_node_ptr,
              unsigned idx)
 {
     const H5AC_class_t *child_class;                            /* Pointer to child node's class info */
-    haddr_t             left_addr, right_addr;                  /* Addresses of left & right child nodes */
+    haddr_t             left_addr = 0, right_addr = 0;          /* Addresses of left & right child nodes */
```

Comment:
Totally the right direction, but the invalid value to use for initializing haddr_t variables is HADDR_UNDEF.